src/main/java/hudson/remoting/Util.java

```diff
@@ -131,4 +126,35 @@ static URLConnection openURLConnection(URL url) throws IOException {
         }
         return con;
     }
+
+    static InetSocketAddress getResolvedHttpProxyAddress(String host, int port) throws IOException {
+        InetSocketAddress targetAddress = null;
+        Iterator<Proxy> proxies = ProxySelector.getDefault().select(URI.create(String.format("http://%s:%d", host, port))).iterator();
+        while (targetAddress == null && proxies.hasNext()) {
+            Proxy proxy = proxies.next();
+            if(proxy.type() == Proxy.Type.HTTP) {
+                final SocketAddress address = proxy.address();
+                if (!(address instanceof InetSocketAddress)) {
+                    System.err.println("Unsupported proxy address type " + (address != null ? address.getClass() : "null"));
+                    continue;
+                }
+                InetSocketAddress proxyAddress = (InetSocketAddress) address;
+                if(proxyAddress.isUnresolved())
+                    proxyAddress = new InetSocketAddress(proxyAddress.getHostName(), proxyAddress.getPort());
+                targetAddress = proxyAddress;
+            }
+        }
+        if(targetAddress == null) {
+            String httpProxy = System.getenv("http_proxy");
+            if(httpProxy != null) {
+                try {
+                    URL url = new URL(httpProxy);
+                    targetAddress = new InetSocketAddress(url.getHost(), url.getPort());
+                } catch (MalformedURLException e) {
+                    System.err.println("Not use http_proxy environment variable which is invalid: "+e.getMessage());
```

Comment:
Is was using the same logging as https://github.com/hypnoce/remoting/blob/master/src/main/java/hudson/remoting/Util.java#L121.